subsys/bluetooth/services/fast_pair/fp_storage/fp_storage_ak.c

```diff
@@ -405,9 +735,250 @@ int fp_storage_ak_save(const struct fp_account_key *account_key)
 			"order may change at reboot.");
 	}
 
+	if (IS_ENABLED(CONFIG_BT_FAST_PAIR_STORAGE_AK_BOND) && ak_overwritten) {
+		/* Account Key overwritten. Remove bonds releted with overwritten Account Key. */
+		bonds_with_invalid_id_remove_completely();
+	}
+
+	return 0;
+}
+
+static int ak_id_get(uint8_t *id, const struct fp_account_key *account_key)
```

Comment:
Maybe we could return `id` through return value (`BOND_AK_ID_UNKNOWN` could denote invalid ID)?

Reply:
It would be confusing as BOND_AK_ID_UNKNOWN (or FP_BOND_INFO_AK_ID_UNKNOWN now) is reserved for bond entries that don't have AK associated to them yet. I'll leave it as is.